test/test_gpu3.cpp

```diff
@@ -9782,33 +9786,49 @@ TEST_F(NVFuserTest, FusionCrossGridInnerReductionSplitGridIteration_CUDA) {
   fusion.addInput(t0);
   fusion.addOutput(t1);
 
-  auto options =
-      at::TensorOptions().dtype(data_type_to_aten(dtype)).device(at::kCUDA, 0);
-  at::Tensor aten_input = at::randn(input_shape, options);
+  // Reference result is calculated by aten in double precision.
+  // Estimated_gmem is 17.18 GBytes, skip if not enough memory.
+  size_t n_elements = reduction_size * iteration_size + iteration_size * 2;
+  size_t estimated_gmem = n_elements * dataTypeSize(dtype);
+  size_t device_free, device_total;
+  cudaMemGetInfo(&device_free, &device_total);
+  if (estimated_gmem > device_free) {
+    GTEST_SKIP() << "Skipping test due to limited GPU memory. Requested: "
+                 << estimated_gmem / 1e9 << " GBytes"
+                 << ", device_free: " << device_free / 1e9 << " GBytes"
+                 << ", device_total: " << device_total / 1e9 << " GBytes";
+  } else {
```

Comment:
```suggestion
  if (estimated_gmem > device_free) {
    GTEST_SKIP() << "Skipping test due to limited GPU memory. Requested: "
                 << estimated_gmem / 1e9 << " GBytes"
                 << ", device_free: " << device_free / 1e9 << " GBytes"
                 << ", device_total: " << device_total / 1e9 << " GBytes";
  }
```

And remove `}` at line 9828

Reply:
changed.